packages/ssx-sdk/src/ssx.ts

```diff
@@ -61,6 +65,20 @@ export class SSX {
 
     try {
       this.session = await this.connection.signIn();
+      let ens = {};
+      if (
+        this.config.resolveEns && (
+          typeof this.config.resolveEns === 'boolean' ||
+          !this.config.resolveEns?.resolveOnServer
+        )
+      ) {
+        let resolveEnsOpts;
+        if (typeof this.config.resolveEns === 'object') {
+          resolveEnsOpts = this.config.resolveEns.resolve;
+        }
+        ens = await this.resolveEns(this.session.address, resolveEnsOpts);
+        this.session.ens = ens;
+      }
```

Comment:
I had a hard time visually parsing this block to understand what the conditions were. Can you check my understanding that this is equivalent code? If so can we simplify?

```ts
let resolveEns = !!this.config.resolveEns && this.config.resolveEns;

try {
  this.session = await this.connection.signIn();

  if (resolveEns?.resolveOnServer) {
    this.session.ens = await this.resolveEns(this.session.address, resolveEns.resolve);
  }

  return this.session;
}
```

Reply:
Yes, I think this accomplishes the same thing

Reply:
It doesn't, but I tried to simplify this code block on [db50a0c](https://github.com/spruceid/ssx/pull/19/commits/db50a0cb3b58665eb2a0cae6936e8cb5cac18cea)